lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module

```diff
@@ -158,6 +158,26 @@ function dosomething_northstar_get_northstar_user($drupal_id) {
 }
 
 /**
+ * Get a Northstar user for a given index & type. Useful for non Drupal ID lookups.
+ * @param  string $index_type The type of index to query for (Email, Facebook, Mobile, _id)
+ * @param  string $index The index to query for
+ * @return object|null
+ */
+function dosomething_northstar_get_user_by_custom_index($index_type, $index) {
```

Comment:
@angaither didnt test locally, but what I did was

```
function dosomething_northstar_get_northstar_user($id, $type = 'drupal_id') {
  $northstar_user_info = NULL;

  if ($type === 'drupal_id') {
    $northstar_user_info = cache_get('northstar_user_info_' . $id, 'cache_dosomething_northstar');
  }

 ...
```

and then for setting the cache 

```
cache_set('northstar_user_info_' . $northstar_user_info->drupal_id, $northstar_user_info, 
```

gonna try it out now